VAULT-45511: Add bounded event queue configuration documentation (#2717)
hashicorp/web-unified-docs: content/vault/v1.21.x/content/docs/concepts/events.mdx

As a result, event notifications **may** still be sent for a few minutes after a token is
revoked or a policy is deleted.

## Bounded event queues

The event notification system supports configurable bounded queues for event subscribers to prevent resource exhaustion in deployments with high subscriber counts.

To enable bounded event queues, set the `VAULT_EVENT_NOTIFICATIONS_BOUNDED_QUEUE_SIZE` environment variable to a positive integer (e.g., 16) on your Vault server. This configures buffered channels of that size (maximum 1000) for each subscriber:

> ```shell

Reviewer: Use `shell-session` instead of `shell`

> VAULT_EVENT_NOTIFICATIONS_BOUNDED_QUEUE_SIZE=16

Reviewer: For style consistency:

```suggestion
$ export VAULT_EVENT_NOTIFICATIONS_BOUNDED_QUEUE_SIZE=16
```

```

> When bounded queues are enabled, subscribers may miss events if the queue fills up faster than events can be consumed. The default value is 0 (unbounded) for backward compatibility.

Reviewer: Avoid using passive voice ([style guide](https://github.com/hashicorp/web-unified-docs/blob/main/docs/style-guide/top-12.md#write-in-active-voice)):

```suggestion
When you enable bounded queues, subscribers may miss events if the queue fills up faster than Vault can process them. The default value is 0 (unbounded) for backward compatibility.
```
